ios/Classes/FlutterPlayer.swift

```diff
@@ -55,37 +55,43 @@ extension FlutterPlayer: FlutterStreamHandler {
 
 private extension FlutterPlayer {
     func handleMethodCall(call: FlutterMethodCall, result: @escaping FlutterResult) {
-        let methodCallResult = handleMethodCall(call: call)
-        result(methodCallResult)
+        do {
+            let methodCallResult = try handleMethodCall(call: call)
+            result(methodCallResult)
+        } catch let bitmovinError as BitmovinError {
+            result(FlutterError.from(bitmovinError))
+        } catch {
+            result(
+                FlutterError.general(
+                    "Error while executing method call \"\(call.method)\": \(error.localizedDescription)"
+                )
+            )
+        }
     }
 
     // swiftlint:disable:next cyclomatic_complexity function_body_length
-    func handleMethodCall(call: FlutterMethodCall) -> Any? {
+    func handleMethodCall(call: FlutterMethodCall) throws -> Any? {
         guard let arguments = Helper.methodCallArguments(call.arguments) else {
-            return FlutterError()
+            throw BitmovinError.parsingError("Could not parse arguments for \(call.method)")
         }
 
         switch (call.method, arguments) {
         case (Methods.loadWithSourceConfig, .json(let sourceConfigJson)):
-            if let (sourceConfig, metadata) = Helper.sourceConfig(sourceConfigJson) {
-                let sourceMetadata = MessageDecoder.toNative(
-                    type: FlutterSourceMetadata.self,
-                    from: sourceConfigJson["analyticsSourceMetadata"]
+            if let flutterSourceConfig = Helper.sourceConfig(sourceConfigJson) {
+                handleLoadSource(
+                    flutterSource: FlutterSource(
+                        sourceConfig: flutterSourceConfig,
+                        remoteControl: nil
```

Comment:
Is it desired that the `SourceConfig` load option does not support a different cast source? 🤔

Reply:
Yes. The problem here is that the remote source is defined on `Source` level. If just a `SourceConfig` is provided, there is no way to configure the remote source. See changes to `source.dart` in this PR. This is aligned with the API that was implemented for React Native